File: 2024/March/_16/C/exerc2_inverte.c
#include <stdio.h>

void inverte(int v[], int total);
void inverteRec(int v[], int pos, int total);

void inverte(int v[], int total)
{
    inverteRec(v, 0, total);
}

void inverteRec(int v[], int pos, int total)
{
    if(pos == total/2)
        return;
    int temp = v[pos];
    v[pos] = v[total-pos-1];
    v[total-pos-1] = temp;
    inverteRec(v, pos+1, total);
}

int main()
{
    int vet[] = { 1, 2, 3, 4, 5, 6, 7, 8 };
    for (int i = 0; i < 8; i++)
        printf("%d ", vet[i]);
    printf("\n");
    inverte(vet,8);
    for (int i = 0; i < 8; i++)
        printf("%d ", vet[i]);
    printf("\n");
}

File: 2024/March/_16/Exerc2_Inverte.java
package March._16;
public class Exerc2_Inverte {

    public static void inverte(int[] v)
    {
        inverteRec(v, 0);
    }

    public static void inverteRec(int[] v, int pos)
    {
        if(pos == v.length/2)
            return;
        int temp = v[pos];
        v[pos] = v[v.length-pos-1];
        v[v.length-pos-1] = temp;
        inverteRec(v, pos+1);
    }

    public static void main(String[] args) {

        int vet[] = { 1, 2, 3, 4, 5, 6, 7, 8 };
        for (int i: vet)
            System.out.printf("%d ", i);
        System.out.println();
        inverte(vet);
        for (int i: vet)
            System.out.printf("%d ", i);
        System.out.println();

    }
}
